src/service/users/index.js

```diff
@@ -1,40 +1,26 @@
-import { executeQuery } from '../subgraph';
-import { config } from '../../config';
+import { executeQuery } from "../subgraph";
+import { config } from "../../config";
 
 function getUniqueUserCountByChain(chainId, version) {
     return new Promise(async (resolve, reject) => {
         try {
-            let totalCount = 0;
-            if(!chainId) {
-                return reject("Invalid chainId passed as parameter. Please check the inputs");
+            if (!chainId) {
+                return reject(
+                    "Invalid chainId passed as parameter. Please check the inputs"
+                );
             }
-
-            let endOfData = false;
-            let skip = 0;
-            while(!endOfData) {
-
-                let query = `query {
-                    uniqueWallets(first: 1000, skip: ${skip}) {
-                        id
-                    }
-                }`
-                skip += 1000;
-                let data = await executeQuery(chainId, query, version);
-                if(data && data.data && data.data.uniqueWallets) {
-                    let aggregateData = data.data.uniqueWallets;
-                    if(aggregateData) {
-                        totalCount += aggregateData.length;
-                    }
-                    if(aggregateData.length == 0) {
-                        endOfData = true;
-                    }
-                } else {
-                    console.log(`No data returned from graph while getting unique user count for chain id ${chainId}`);
-                    endOfData = true;
+            const query = `{
+                uniqueWalletCounts{
```

Comment:
Filter this query by  `where { id: 1}`
and in the result just pick the count field... no need to pick the last element in the array

Reply:
awesome, will change!